sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py

```diff
@@ -646,8 +657,9 @@ def dead_letter(self, reason=None, description=None):
         self._check_live(MESSAGE_DEAD_LETTER)
 
         details = {
-            MGMT_REQUEST_DEAD_LETTER_REASON: str(reason) if reason else "",
-            MGMT_REQUEST_DEAD_LETTER_DESCRIPTION: str(description) if description else ""}
+            RECEIVER_LINK_DEAD_LETTER_REASON: reason,
+            RECEIVER_LINK_DEAD_LETTER_DESCRIPTION: description
+        }
 
         self._settle_message(MESSAGE_DEAD_LETTER, dead_letter_details=details)
```

Comment:
Structural nit:  settle_message does both receiver and mgmt based dead-lettering.  The way we're populating this via the full dictionary results in having to extract the properties using the RECEIVER_LINK consts in the mgmt flow, and requires defining receiver-specific structure in this outer scope.  I'd likely lean towards passing both dead_letter_description and _reason as explicit variables into the top level _settle_message function, and building the object as appropriate within each internal flow.

Feel free to counter-argue this, you have a great track record of having preempted many of the silly things I notice :P

Reply:
I agree upon this, updated -- previously I was just too lazy to do the update ><